inference-engine/src/mkldnn_plugin/nodes/gather_elements.cpp

```diff
@@ -0,0 +1,153 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "base.hpp"
+
+#include <string>
+#include <vector>
+#include "ie_parallel.hpp"
+#include "jit_generator.hpp"
+#include "common/cpu_memcpy.h"
+#include <mkldnn_types.h>
+#include <chrono>
+
+namespace InferenceEngine {
+namespace Extensions {
+namespace Cpu {
+
+class GatherElementsImpl: public ExtLayerBase {
+public:
+    explicit GatherElementsImpl(const CNNLayer* layer) : strideAx1Diff_(0) {
+        errorPrefix_ = std::string("Layer GatherElements with name '") + layer->name + "'";
+
+        if (layer->insData.size() != 2 || layer->outData.size() != 1)
+            THROW_IE_EXCEPTION << errorPrefix_ << " has invalid number of input/output edges.";
+
+        auto inputData = layer->insData[dataIndex_].lock();
+        auto indices = layer->insData[indicesIndex_].lock();
+        if (!inputData || !indices)
+            THROW_IE_EXCEPTION << errorPrefix_ << " has nullable inputs.";
+
+        const auto& dataDims = inputData->getTensorDesc().getDims();
+        const auto& indicesDims = indices->getTensorDesc().getDims();
+        if (dataDims.size() != indicesDims.size())
+            THROW_IE_EXCEPTION << errorPrefix_ << " has invalid input shapes. Inputs 'Data' and 'Indices' must have equal ranks.";
+
+        Precision dataPrecision = inputData->getTensorDesc().getPrecision();
+        if (dataPrecision.size() != sizeof(PrecisionTrait<Precision::I32>::value_type) &&
+                dataPrecision.size() != sizeof(PrecisionTrait<Precision::I16>::value_type) &&
+                dataPrecision.size() != sizeof(PrecisionTrait<Precision::I8>::value_type)) {
+            THROW_IE_EXCEPTION << errorPrefix_ << " has unsupported 'inputData' input precision: " << dataPrecision;
+        }
+
+        Precision indicesPrecision = indices->getTensorDesc().getPrecision();
+        if (indicesPrecision != Precision::I32) {
+            THROW_IE_EXCEPTION << errorPrefix_ << " has unsupported 'indices' input precision: " << indicesPrecision;
+        }
+
+        dataTypeSize_ = dataPrecision.size();
+
+        int axis = layer->GetParamAsInt("axis", 0);
```

Comment:
The spec says that axis attribute is mandatory so we shouldn't specify default value on our own.

Reply:
Fixed